R-package/R/lgb.Booster.R

```diff
@@ -671,19 +699,29 @@ predict.lgb.Booster <- function(object, data,
 #'                    early_stopping_rounds = 10)
 #' lgb.save(model, "model.txt")
 #' load_booster <- lgb.load("model.txt")
+#' load_booster <- lgb.load(model_str)
 #' }
 #' 
 #' @rdname lgb.load
 #' @export
 lgb.load <- function(filename){
   
   # Check if file name is character or not
-  if (!is.character(filename)) {
+  if (supplied(filename) && !is.null(filename) && !is.character(filename)) {
     stop("lgb.load: filename should be character")
   }
-  
+    
+  if (supplied(model_str) && !is.null(model_str) && !is.character(model_str)) {
+    stop("lgb.load: model_str should be character")
+  }
+    
+  if (!supplied(filename) && !supplied(model_str)) {
+    stop("lgb.load: either filename or model_str must be supplied")
+  }
+    
   # Return new booster
-  Booster$new(modelfile = filename)
+  if (supplied(filename)) Booster$new(modelfile = filename)
```

Comment:
You can use `file.exists` to check for `filename` file existence.

If file exists, you can load as usual from file. Otherwise, you can attempt to load model from string. You may separate loading from string and loading from model if needed, it requires a different argument apparently in your implementation (model_str).